src/main/java/com/google/api/codegen/metacode/InitCodeGenerator.java

```diff
@@ -206,6 +202,12 @@ private InitCodeLine generateCodeInit(
                   + initFieldStructure);
         }
         initValueConfig = initValueConfig.withInitialValue(validatedValue);
+      } else if (typeRef.isPrimitive()
```

Comment:
Yes, I moved the check for hasInitialValue() first so that it is possible to set the initial value for the page token in tests, when shouldGenerateTestValue() is true